lib/pbench/test/unit/agent/task/test_results_push.py

```diff
@@ -28,32 +28,20 @@ class TestResultsPush:
     def add_http_mock_response(
         status_code: HTTPStatus = None, message: Optional[Union[str, Dict]] = None
     ):
+        parms = {}
         if status_code:
-            if message is None:
-                responses.add(
-                    responses.PUT,
-                    f"{TestResultsPush.URL}/upload/{os.path.basename(tarball)}",
-                    status=status_code,
-                )
-            elif isinstance(message, dict):
-                responses.add(
-                    responses.PUT,
-                    f"{TestResultsPush.URL}/upload/{os.path.basename(tarball)}",
-                    status=status_code,
-                    json=message,
-                )
-            else:
-                responses.add(
-                    responses.PUT,
-                    f"{TestResultsPush.URL}/upload/{os.path.basename(tarball)}",
-                    status=status_code,
-                    body=message,
-                )
-        else:
-            responses.add(
-                responses.PUT,
-                f"{TestResultsPush.URL}/upload/{os.path.basename(tarball)}",
-            )
+            parms["status"] = status_code
+
+        if isinstance(message, dict):
+            parms["json"] = message
+        elif isinstance(message, (str, Exception)):
+            parms["body"] = message
```

Comment:
I probably should have deleted the `Exception` case as well since I didn't understand it in the first place.

Reply:
> What would that mean?

It's a cool feature (e.g., for testing):  it causes the request to raise the specified exception instead of returning with a response (e.g., it's handy for [testing the handling of `ConnectionError`](https://github.com/distributed-system-analysis/pbench/blob/af615a24992caf9a08743cf7f8984881035c92b0/lib/pbench/test/unit/agent/task/test_results_push.py#L38)).

Reply:
Ah; indeed. I think I may even have used that. The fact that I'd forgotten points out what an (admittedly convenient, but) extremely bizarre overloading this is.